app/domain/authentication/authn_k8s/extract_k8s_resource_restrictions.rb

```diff
@@ -0,0 +1,71 @@
+require 'command_class'
+
+module Authentication
+  module AuthnK8s
+
+    ExtractK8sResourceRestrictions = CommandClass.new(
+      dependencies: {
+        extract_resource_restrictions: Authentication::ResourceRestrictions::ExtractResourceRestrictions.new,
+        resource_restrictions_class:   Authentication::ResourceRestrictions::ResourceRestrictions,
+        logger:                        Rails.logger
+      },
+      inputs:   %i(authenticator_name service_id role_name account)
+    ) do
+
+      def call
+        extract_resource_restrictions
+      end
+
+      private
+
+      # Extracts resource restrictions for k8s hosts. It can be defined in the
+      # annotations or as part of the host ID itself.
+      # If the restrictions are in the annotations, then they will not be
+      # extracted from the host ID.
+      def extract_resource_restrictions
+        # Extract resource restrictions from host annotations.
+        resource_restrictions = @extract_resource_restrictions.call(
+          authenticator_name: @authenticator_name,
+          service_id:         @service_id,
+          role_name:          @role_name,
+          account:            @account
+        )
+
+        has_restrictions_besides_container_name = !resource_restrictions.except(Restrictions::AUTHENTICATION_CONTAINER_NAME).empty?
```

Comment:
Ah!  Here's that concept of "restriction type" (from one of my other comments) popping up again!  I often find that when I want to name things with long phrases like this, there is a concept I haven't found yet.  I'm not sure what it is in the case yet.

Reply:
> I'm not sure what it is in the case yet.

Me neither... 🤔